app/assess/routes.py

```diff
@@ -467,3 +463,89 @@ def continue_assessment(application_id):
         section="NA",
         page_to_render="continue_assessment.html",
     )
+
+
+@assess_bp.route("/application/<application_id>/export", methods=["GET"])
+@login_required(roles_required=["LEAD_ASSESSOR"])
+def generate_doc_list_for_download(application_id):
+    current_app.logger.info(
+        f"Generating docs for application id {application_id}"
+    )
+    state = get_banner_state(application_id)
+    short_id = state.short_id[-6:]
+    latest_flag = get_latest_flag(application_id)
+    display_status = determine_display_status(
+        state.workflow_status, latest_flag
+    )
+
+    fund = get_fund(state.fund_id)
+    application_json = get_application_json(application_id)
+    supporting_evidence = get_files_for_application_upload_fields(
+        application_id=application_id,
+        short_id=short_id,
+        application_json=application_json,
+    )
+    application_answers = (
+        "Application answers",
+        url_for(
+            "assess_bp.download_application_answers",
+            application_id=application_id,
+            short_id=short_id,
+        ),
+    )
+
+    return render_template(
+        "contract_downloads.html",
+        application_id=application_id,
+        fund_name=fund.name,
+        state=state,
+        application_answers=application_answers,
+        supporting_evidence=supporting_evidence,
+        display_status=display_status,
+    )
+
+
+@assess_bp.route("/application/<application_id>/export/<short_id>/answers.txt")
+@login_required(roles_required=["LEAD_ASSESSOR"])
+def download_application_answers(application_id: str, short_id: str):
+    current_app.logger.info(
+        f"Generating application Q+A download for application {application_id}"
+    )
+    application_json = get_application_json(application_id)
+    qanda_dict = extract_questions_and_answers_from_json_blob(
+        application_json["jsonb_blob"]
+    )
+    text = generate_text_of_application(qanda_dict)
+
+    return download_file(text, "text/plain", f"{short_id}_answers.txt")
+
+
+@assess_bp.route(
+    "/application/<application_id>/export/<file_name>",
+    methods=["GET"],
+)
+@assess_bp.route(
+    "/application/<application_id>/export/<short_id>/<file_name>",
+    methods=["GET"],
+)
```

Comment:
Could we combine these into one route with short_id as an optional query parameter?

Reply:
have combined to use 1 route with short_id as a query parameter 👍🏾